src/FixMath.h

```diff
@@ -742,27 +717,32 @@ class SFix
       @return The result of the addition as a SFix.
   */
   template<int8_t _NI, int8_t _NF, uint64_t _RANGE>
-  SFix<FixMathPrivate::neededSNIExtra(FixMathPrivate::FM_max(NI,_NI),FixMathPrivate::FM_max(NF,_NF),FixMathPrivate::rangeAdd(NF,_NF,RANGE,_RANGE)),FixMathPrivate::FM_max(NF, _NF), FixMathPrivate::rangeAdd(NF,_NF,RANGE,_RANGE)> operator+ (const SFix<_NI,_NF,_RANGE>& op) const
+  constexpr typename SFix<FixMathPrivate::FM_max(NI,_NI),FixMathPrivate::FM_max(NF,_NF),FixMathPrivate::rangeAdd(NF,_NF,RANGE,_RANGE)>::SFixNIadj_t operator+ (const SFix<_NI,_NF,_RANGE>& op) const
   {
-    constexpr uint64_t new_RANGE = FixMathPrivate::rangeAdd(NF,_NF,RANGE,_RANGE);
-    constexpr int8_t new_NI = FixMathPrivate::neededSNIExtra(FixMathPrivate::FM_max(NI,_NI),FixMathPrivate::FM_max(NF,_NF),new_RANGE);
-    constexpr int8_t new_NF = FixMathPrivate::FM_max(NF, _NF);
-    typedef typename IntegerType<FixMathPrivate::sBitsToBytes(new_NI+new_NF)>::signed_type return_type;
-    SFix<new_NI,new_NF> left(*this);
-    SFix<new_NI,new_NF> right(op);
-
-    return_type tt = return_type(left.asRaw()) + right.asRaw();
-    return SFix<new_NI, new_NF, new_RANGE>(tt,true);
+    using namespace FixMathPrivate;
+    typedef typename SFix<FM_max(NI,_NI),FM_max(NF,_NF),rangeAdd(NF,_NF,RANGE,_RANGE)>::SFixNIadj_t worktype;
+    return worktype(worktype(*this).asRaw() + worktype(op).asRaw(), true);
   }
-  
+
+  /** Addition between a SFix and a UFix. Safe.
```

Comment:
Did I forgot this one initially? Or maybe I put it as a free function

Reply:
This and the one or two others used to be free functions. I brought them into the classes to avoid extra friend declarations (for ```U/SFixNIadj_t```).